website/siteConfig.js

```diff
@@ -50,6 +50,7 @@ const siteConfig = {
     'https://buttons.github.io/buttons.js',
     'https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js',
     '/js/code-blocks-buttons.js',
+    '/js/vanilla-back-to-top.min.js'
```

Comment:
Oh yes you're right. So what do you propose for that ? 
Maybe, the better way is to use the npm package ? (https://www.npmjs.com/package/vanilla-back-to-top)

Reply:
Do I need to put the new option to the basics siteCOnfig.js ? Because it's optional. 
I don't know the politic of this project for this kink of things.
By default we put all optionals features to true ?

Reply:
There are many ways to do this. Through npm package is doable. 
Hardcoding the script at Head.js is also doable (but not preferable).
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/core/Head.js#L141-L150

As for putting it on basic `siteConfig`, i think there is no need for that if it's optional. Most user would be updating their docusaurus version anyway. The next question is just whether we make this optional or not. 

I'll leave it to the next reviewer that has the authority to decide 😄. I'm just adding my comment on the code here

Reply:
There are many ways to do this. Through npm package is doable. 
Hardcoding the script at Head.js is also doable (but not preferable).
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/core/Head.js#L141-L150

As for putting it on basic `siteConfig`, i think there is no need for that if it's optional. Most user would be updating their docusaurus version & retaining their config. All optional features are not defaulted to true.

Anyway, I'll leave the rest to the next reviewer 😄. I'm just adding my comment on the code here

Reply:
Ok, so I wait some reviewers to do more commits :)

Reply:
I do recommend you to change the thing I pointed out though, because putting it here won't make any user be able to use the feature

Reply:
Hum, I'm little lost on the architecture. 
If I put the package from npm, where can I put it. 
My first reaction is to put in the website/package.json, but we still have the same issue ?

Reply:
It doesn't seems to work as a package : 

![image](https://user-images.githubusercontent.com/2819925/40878164-2dbdabf4-668d-11e8-8383-f5dc5495cb85.png)
![image](https://user-images.githubusercontent.com/2819925/40878165-37d45368-668d-11e8-937c-2557f4b77a0b.png)
![image](https://user-images.githubusercontent.com/2819925/40878168-4431c208-668d-11e8-8a60-daf21c79db40.png)
![image](https://user-images.githubusercontent.com/2819925/40878172-4f293998-668d-11e8-9c64-74c7f9d6c4de.png)

Reply:
Hmm.. I've never used the package before. See if embedding the script works
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/core/Head.js#L141-L150

Reply:
You mean that I copy/paste the script here ? 
I don't thinks it's a good idea for mainteners

Reply:
I see
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/core/Head.js#L97-L102

Otherwise, we can put in docs if they want this feature they should put that script. WDYT ?

Reply:
Hmmm.. Does it work ?

Reply:
Yes !

I try to put a js folder in lib/static, but the js folder is not serve I suppose...
In css I can put every file I want, I can get them.

Reply:
Can you make the change and push it as commit ? I want to try it again. Actually this feature is good. I'm not expert at UI hahah

Reply:
Done

Reply:
Try removing `/js/vanilla-back-to-top.min.js' from scripts

Reply:
Sorry, I forget to commit this file ;)

And I try the problem with the static file. So now the lib is in /lib/static/js folder.
So we fetch from the site and not from the unpkg site :)